python/pyarrow/tensorflow/plasma_op.cc

```diff
@@ -0,0 +1,381 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "tensorflow/core/framework/device_base.h"
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/framework/shape_inference.h"
+#include "tensorflow/core/platform/logging.h"
+#include "tensorflow/core/platform/mutex.h"
+#include "tensorflow/stream_executor/device_memory.h"
+#include "tensorflow/stream_executor/event.h"
+#include "tensorflow/stream_executor/stream.h"
+
+#ifdef GOOGLE_CUDA
+#include "tensorflow/core/common_runtime/gpu/gpu_event_mgr.h"
+#include "tensorflow/core/platform/stream_executor.h"
+#endif
+
+#include "arrow/adapters/tensorflow/convert.h"
+#include "arrow/io/memory.h"
+#include "arrow/python/arrow_to_python.h"
+#include "arrow/python/python_to_arrow.h"
+#include "arrow/tensor.h"
+#include "plasma/client.h"
+
+
+using namespace tensorflow;  // NOLINT
+
+using ArrowStatus = arrow::Status;
+using CPUDevice = Eigen::ThreadPoolDevice;
+using GPUDevice = Eigen::GpuDevice;
+
+static plasma::PlasmaClient client_;
+static bool connected_ = false;
+static mutex mu_;
+
+using Event = perftools::gputools::Event;
+using Stream = perftools::gputools::Stream;
+
+// NOTE(zongheng): for some reason using unique_ptr or shared_ptr results in
```

Comment:
Who is zongheng? I didn't see him/her in the commits

Reply:
zongheng is @concretevitamin (mentioned in the PR message), he is a PhD student at the RISELab

Reply:
I originally authored this code in a private repo.  Then @pschafhalter and @pcmoritz took it over to productionize/push it into arrow codebase.